Sources/Build/BuildDelegate.swift

```diff
@@ -0,0 +1,223 @@
+	private let diagnostics: DiagnosticsEngine
+	private let outputStream: OutputByteStream
+
+	public init(diagnostics: DiagnosticsEngine, outputStream: OutputByteStream = stdoutStream) {
+		self.diagnostics = diagnostics
+		self.outputStream = outputStream
+	}
+
+    /// The FileSystem to use, if any.
+    ///
+    /// This is currently very limited.
+    public var fs: llbuildSwift.FileSystem? {
+        return nil
+    }
+
+    /// Called in response to requests for new tools.
+    ///
+    /// The client should return an appropriate tool implementation if recognized.
+    public func lookupTool(_ name: String) -> Tool? {
+        return nil
+    }
+    
+    /// Called to report any form of command failure.
+    ///
+    /// This can may be called to report the failure of a command which has
+    /// executed, but may also be used to report the inability of a command to
+    /// run. It is expected to be used by the client in making decisions with
+    /// regard to cancelling the build.
+    public func hadCommandFailure() {
+    }
+    
+    /// Called to report an unassociated diagnostic from the build system.
+    public func handleDiagnostic(_ diagnostic: llbuildSwift.Diagnostic) {
+    }
+
+    /// Called when a command has changed state.
+    ///
+    /// The system guarantees that any commandStart() call will be paired with
+    /// exactly one \see commandFinished() call.
+    public func commandStatusChanged(_ command: llbuildSwift.Command, kind: CommandStatusKind) {
+    }
+
+    /// Called when a command is preparing to start.
+    ///
+    /// The system guarantees that any commandStart() call will be paired with
+    /// exactly one \see commandFinished() call.
+    public func commandPreparing(_ command: llbuildSwift.Command) {
+    }
+
+    /// Called when a command has been started.
+    ///
+    /// The system guarantees that any commandStart() call will be paired with
+    /// exactly one \see commandFinished() call.
+    public func commandStarted(_ command: llbuildSwift.Command) {
+        guard command.shouldShowStatus else { return }
+		outputStream <<< command.description <<< "\n"
+    }
+    
+    /// Called to allow the delegate to skip commands without cancelling their
+    /// dependents. See llbuild's should_command_start.
+    public func shouldCommandStart(_ command: llbuildSwift.Command) -> Bool {
+        return true
+    }
+
+    /// Called when a command has been finished.
+    public func commandFinished(_ command: llbuildSwift.Command, result: CommandResult) {
+    }
+
+    /// Called to report an error during the execution of a command.
+    public func commandHadError(_ command: llbuildSwift.Command, message: String) {
+        diagnostics.emit(data: LLBuildCommandErrorDiagnostic(message: message))
+    }
+
+    /// Called to report a note during the execution of a command.
+    public func commandHadNote(_ command: llbuildSwift.Command, message: String) {
+        outputStream <<< message <<< "\n"
```

Comment:
Why is this one written to output stream directly and others are diagnostics? That seems wrong.